firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalStore.java

```diff
@@ -778,13 +782,52 @@ public void saveNamedQuery(NamedQuery namedQuery, ImmutableSortedSet<DocumentKey
         "Get named query", () -> bundleCache.getNamedQuery(queryName));
   }
 
-  public void configureIndices(List<FieldIndex> fieldIndices) {
+  @VisibleForTesting
+  Collection<FieldIndex> getFieldIndexes() {
+    return persistence.runTransaction("Get indexes", () -> indexManager.getFieldIndexes());
+  }
+
+  public void configureFieldIndexes(List<FieldIndex> newFieldIndexes) {
     persistence.runTransaction(
-        "Configure indices",
+        "Configure indexes",
         () -> {
-          // TODO(indexing): Disable no longer active indices
-          for (FieldIndex fieldIndex : fieldIndices) {
-            indexManager.addFieldIndex(fieldIndex);
+          List<FieldIndex> existingIndexes = new ArrayList<>(indexManager.getFieldIndexes());
+          Collections.sort(existingIndexes, FieldIndex.SEMANTIC_COMPARATOR);
+          Iterator<FieldIndex> existingIt = existingIndexes.iterator();
+          @Nullable FieldIndex existingValue = advanceIterator(existingIt);
+
+          List<FieldIndex> updatedIndexes = new ArrayList<>(newFieldIndexes);
+          Collections.sort(updatedIndexes, FieldIndex.SEMANTIC_COMPARATOR);
+          Iterator<FieldIndex> updatedIt = updatedIndexes.iterator();
+          @Nullable FieldIndex updatedValue = advanceIterator(updatedIt);
+
+          while (existingValue != null || updatedValue != null) {
+            boolean deleted = false;
```

Comment:
I think they can be named `deleteExistingIndex` and `writeNewIndex`?

Reply:
I actually realized yesterday that I need to re-use this functionality in a follow up PR, so I changed this method to be more generic and moved it to Util.